RecoMuon/GlobalTrackingTools/src/MuonTrackingRegionBuilder.cc

```diff
@@ -1,281 +1,281 @@
+  vertexCollectionToken = iC.consumes<reco::VertexCollection>(par.getParameter<edm::InputTag>("vertexCollection"));
+
+  // Input muon collection
+  inputCollectionToken = iC.consumes<reco::TrackCollection>(par.getParameter<edm::InputTag>("input"));
+}
+
+
+//
+// Member function to be compatible with TrackingRegionProducerFactory: create many ROI for many tracks
+//
+std::vector<TrackingRegion*> MuonTrackingRegionBuilder::regions(const edm::Event& ev, const edm::EventSetup& es) const {
+	
+  std::vector<TrackingRegion*> result;
+
+  edm::Handle<reco::TrackCollection> tracks;
+  ev.getByToken(inputCollectionToken, tracks);
+
+  int nRegions = 0;
+  for (auto it = tracks->cbegin(), ed = tracks->cend(); it != ed && nRegions < theMaxRegions; ++it) {
+    result.push_back(region(*it,ev));
+    nRegions++; 
+  }
+
+  return result;
+
+}
+
+
+//
+// Call region on Track from TrackRef
+//
+RectangularEtaPhiTrackingRegion* MuonTrackingRegionBuilder::region(const reco::TrackRef& track) const {
+  return region(*track);
+}
+
+
+//
+// ToDo: Not sure if this is needed?
+//
+void MuonTrackingRegionBuilder::setEvent(const edm::Event& event) {
+  theEvent = &event;
+}
+
+
+//
+//	Main member function called to create the ROI
+//
+RectangularEtaPhiTrackingRegion* MuonTrackingRegionBuilder::region(const reco::Track& staTrack, const edm::Event& ev) const {
+
+  // get track momentum/direction at vertex
+  const math::XYZVector& mom = staTrack.momentum();
+  GlobalVector dirVector(mom.x(),mom.y(),mom.z());
+  double pt = staTrack.pt();
+
+  // initial vertex position - in the following it is replaced with beamspot/vertexing
+  GlobalPoint vertexPos(0.0,0.0,0.0);
+  // standard 15.9, if useVertex than use error from  vertex
+  double deltaZ = theHalfZ;
+
+  // retrieve beam spot information
+  edm::Handle<reco::BeamSpot> bs;
+  bool bsHandleFlag = ev.getByToken(beamSpotToken, bs);
+
+  // check the validity, otherwise vertexing
+  if ( bsHandleFlag && bs.isValid() && !useVertex ) {
+    vertexPos = GlobalPoint(bs->x0(), bs->y0(), bs->z0());
+    deltaZ = useFixedZ ? theHalfZ : bs->sigmaZ() * theNsigmaDz;
+  } else {
+    // get originZPos from list of reconstructed vertices (first or all)
+    edm::Handle<reco::VertexCollection> vertexCollection;
+    bool vtxHandleFlag = ev.getByToken(vertexCollectionToken, vertexCollection);
+    // check if there exists at least one reconstructed vertex
+    if ( vtxHandleFlag && !vertexCollection->empty() ) {
+      // use the first vertex in the collection and assume it is the primary event vertex 
+      reco::VertexCollection::const_iterator vtx = vertexCollection->begin();
+      if (!vtx->isFake() && vtx->isValid() ) {
+        vertexPos = GlobalPoint(vtx->x(),vtx->y(),vtx->z());
+        deltaZ = useFixedZ ? theHalfZ : vtx->zError() * theNsigmaDz;
+      }
+    }
+  }
+
+  // inizialize to the maximum possible value
+  double deta = 0.4;

```

Comment:
Hi Vincenzo, are these not all already doubles? Or do you suggest changing them all to float?